src/Install/AdminUser.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Install;
+
+use Carbon\Carbon;
+use Illuminate\Hashing\BcryptHasher;
+
+class AdminUser
+{
+    private $username;
+    private $password;
+    private $email;
+
+    public function __construct($username, $password, $email)
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->email = $email;
+
+        $this->validate();
+    }
+
+    public function getUsername()
+    {
+        return $this->username;
+    }
+
+    public function getAttributes(): array
+    {
+        return [
+            'username' => $this->username,
+            'email' => $this->email,
+            'password' => (new BcryptHasher)->make($this->password),
```

Comment:
Is it wise to hardcode the password hasher like this instead of binding it in the container? Some use cases:

- incompatible OS for bcrypt
- newer crypt lib available in Laravel but not supported here
- inability to exchange the hasher from ioc during installation, not sure whether this has any value though 🤷‍♂️

Reply:
I want to keep the installer as decoupled from runtime Flarum as possible. This makes things much easier. When the core changes its default hashing mechanism, the installer will have to change, too.

IMO, we should keep extensions out of the installer. That just adds too much complexity to their lifecycle. They can still change the hashing mechanism at runtime, if they must. (They would have to be able to deal with "legacy" passwords generated by a Flarum without this extension anyway.)